tests/phpunit/tests/media.php

```diff
@@ -3278,6 +3278,51 @@ public function test_wp_get_attachment_image_loading_opt_out_individual() {
 		$this->assertStringNotContainsString( ' loading=', $img );
 	}
 
+	/**
+	 * @ticket 57086
+	 *
+	 * @dataProvider data_test_wp_get_attachment_image_decoding_attr
+	 */
+	public function test_wp_get_attachment_image_decoding_attr( $decoding, $expected ) {
+		if ( 'no value' === $decoding ) {
+			$image = wp_get_attachment_image( self::$large_id, 'thumbnail', false, array() );
+		} else {
+			$image = wp_get_attachment_image( self::$large_id, 'thumbnail', false, array( 'decoding' => $decoding ) );
+		}
+
+		$this->assertSame( $expected, $image );
+	}
+
+	public function data_test_wp_get_attachment_image_decoding_attr() {
+		return array(
+			'default' => array(
+				'decoding' => 'no value',
+				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" decoding="async" loading="lazy" />',
+			),
+			'async'   => array(
+				'decoding' => 'async',
+				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" decoding="async" loading="lazy" />',
+			),
+			'sync'    => array(
+				'decoding' => 'sync',
+				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" decoding="sync" loading="lazy" />',
+			),
+			'auto'    => array(
+				'decoding' => 'auto',
+				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" decoding="auto" loading="lazy" />',
+			),
+			'empty'   => array(
+				'decoding' => '',
+				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" loading="lazy" />',
+			),
+			'false'   => array(
+				'decoding' => false,
+				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" loading="lazy" />',
+			),
```

Comment:
Added a couple more falsey checks and an invalid value.

`=>` will need to be realigned as some of the keys are longer than the existing space allows. I've left that as an exercise for the developer.

```suggestion
			),
			'null'    => array(
				'decoding' => null,
				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" loading="lazy" />',
			),
			'zero'    => array(
				'decoding' => 0,
				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" loading="lazy" />',
			),
			'zero string' => array(
				'decoding' => '0',
				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" loading="lazy" />',
			),
			'zero float' => array(
				'decoding' => 0.0,
				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" loading="lazy" />',
			),
			'invalid'  => array(
				'decoding' => 'invalid',
				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" loading="lazy" />',
			),
```